### ISSUE TITLE
Future of Coding Weekly 2020/06 Week 5

### ISSUE THREAD

**Issue by marianoguerra** [marianoguerra/future-of-coding-weekly#30]
This issue collects resources for the newsletter of the year/month/week that should be in the title

If you want to contribute open the issue and paste a resource you think is worth sharing in the newsletter and comment on the issue of the week.

**Use markdown for formatting**, ideally **a contribution should have a title a link and optionally a description.**

Add emojis to signal videos 🎥 article 📝 demo 🎮 thread 🧵 audio 🎧 twitter 🐦 idea 🧠 podcast 🎙️

Use comment reactions on issue comments to up vote/down vote/whatever each submission the reactions that have a clear sense of positive/negative will be considered to rank the submissions of the week.

**Comment by marianoguerra** [comment]
🎥 [Ruby Conf 12 - Xiki: the Rubyfied Next-Generation Shell by Craig Muth](https://youtu.be/QqOrQN0bxNE) via [Garth Goldwater](https://twitter.com/no_defects) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-22T16:45:48.403Z)

> CANNOT believe I hadn’t heard of Xiki. hard to describe. kind of like a text-base glamorous toolkit or REBL. kind of like a structural interface over the command line. tons of ideas (more info in thread)

**Comment by marianoguerra** [comment]
📝 [Dark and the long term](https://medium.com/darklang/dark-and-the-long-term-2c65ff0baf5e) via **Srini Kadamati** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-23T14:25:39.405Z)

> TL;DR: We’re taking a longer term approach to building Dark. As part of this, we’ve made the difficult decision to shrink Dark’s team, and to change how we build both the product and the company.

**Comment by marianoguerra** [comment]
📝 [THERE IS NO INTERFACE (WITHOUT A USER). A CYBERNETIC PERSPECTIVE ON INTERACTION](https://interfacecritique.net/journal/volume-1/scherffig-there-is-no-interface) via [S.M Mukarram Nainar](https://git.sr.ht/~sm2n/) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-23T15:05:57.409Z)

> “On the surface, the fundamental idea of a direct manipulation interface to a task flies in the face of two thousand years of development of abstract formalisms as a means of understanding and controlling the world. Until very recently, the use of computers has been an activity squarely
> in that tradition. So the exterior of direct manipulation, providing as it does for the direct control of a specific task world, seems somehow atavistic, a return to concrete thinking.”

> Some interesting stuff here, though I'm not sure I buy into the gestalt psychology stuff. This journal seems to have other neat stuff (that I haven't read) too. I saw something on xanadu in the table of contents.

**Comment by marianoguerra** [comment]
🎥 [Scott Kim’s Viewpoint Demo](https://www.youtube.com/watch?v=9G0r7jL3xl8&feature=youtu.be&t=764) via **Jared Windover** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-23T15:21:48.411Z)

> At this section, he’s able to do some modification of the ui of a drawing tool using that tool. Most of the video seemed pretty meh to me (oh a pixel-based drawing tool), but I found this particular segment extremely compelling.

**Comment by marianoguerra** [comment]
📄 [Do Neural Ranking Models Intensify Gender Bias?](https://arxiv.org/pdf/2005.00372.pdf) via [Maeliza](https://www.linkedin.com/in/ma%C3%ABliza-seymour-ab749a36/) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-24T00:39:35.422Z)

> Future of coding should also means more equity. I came accross a paper earlier mentionning that latest NLP algo intensify gender bias in Information Retrival system. Basically, the more performant is the algo the more it is able to infuse the bias of the training dataset. When it comes to NLP to fight gender bias, there are some technics like gender swapping,biais fine tuning or learning gender-neutral embedding exist to mitigate the bias (futher reading - https://arxiv.org/pdf/1906.08976)

**Comment by marianoguerra** [comment]
💻 [Amazon's no code product just launched](https://www.honeycode.aws/) via [Mariano Guerra](https://twitter.com/warianoguerra) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-24T20:36:22.432Z)

> [HN Discussion](https://news.ycombinator.com/item?id=23633110)

**Comment by marianoguerra** [comment]
💬  [Stefan Lesser](https://twitter.com/stefanlesser) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-25T08:28:05.439Z)

> Anybody else here following Apple’s WWDC and having FoC-related thoughts about it?

**Comment by marianoguerra** [comment]
❓ **Aleks Litynski** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-25T17:20:59.452Z)

> 'm trying to write a block based editor (in the style of unreal blueprints and its ilk), but I don't want to let users move nodes freely. I have an algorithm I'm fairly happy with to position the nodes, and now I need to draw edges between them. I'm fairly certain this is NP hard, but that circuit board designers deal with this a lot and have some pretty solid algorithms to do it - https://en.wikipedia.org/wiki/Routing_(electronic_design_automation). Is anyone aware of general purpose libraries that are able to do this?

**Comment by marianoguerra** [comment]
📢 [Python’s adding structural pattern matching](https://www.python.org/dev/peps/pep-0622/) via **korede** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-25T21:15:06.472Z)

**Comment by marianoguerra** [comment]
📝 [Enso Dev Blog — 19th June 2020](https://medium.com/@enso_org/enso-dev-blog-19th-june-2020-335e528d50b) via [Mariano Guerra](https://twitter.com/warianoguerra) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-26T13:08:28.480Z)

> Luna is now called Enso and they are starting from scratch

**Comment by marianoguerra** [comment]
❓ **Spencer Baugh** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/general.html#2020-06-27T02:44:12.497Z)

> Any suggestions on a good front-end library/framework for making drag and drop interactive OOP UIs? for example, a workspace of objects, and clicking a method inside an object then clicking on other objects to pass them as arguments. lots of these exist for specific languages, I'm sure, but if I wanted to make one for my own language, what would I use?

**Comment by marianoguerra** [comment]
🎥 [Future of Coding, Week 10 Update Video](https://www.youtube.com/watch?v=fpKJFDT28LQ&feature=youtu.be) via [Chris Maughan](https://twitter.com/cmaughan) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W5/two-minute-week.html#2020-06-25T12:46:12.003Z)

> This week, an update on scripting the synthesizers I started a week or so ago.

**Comment by marianoguerra** [comment]
https://tinyletter.com/marianoguerra/letters/future-of-coding-weekly-2020-06-week-5-30
